src/Microsoft.AspNetCore.Routing/CompositeEndpointDataSource.cs

```diff
@@ -4,17 +4,20 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Threading;
 using Microsoft.Extensions.Primitives;
 
 namespace Microsoft.AspNetCore.Routing
 {
-    public class CompositeEndpointDataSource : EndpointDataSource
+    public class CompositeEndpointDataSource : EndpointDataSource, IDisposable
```

Comment:
What does disposing actually do here? Like, what's the impact if you don't dispose?